compiler/AST/type.cpp

```diff
@@ -1054,7 +1059,13 @@ void initPrimitiveTypes() {
   dtInt[INT_SIZE_64]                   = createPrimitiveType("int",      "int64_t");
   dtReal[FLOAT_SIZE_64]                = createPrimitiveType("real",     "_real64");
 
-  dtStringC                            = createPrimitiveType("c_string", "c_string" );
+  // The Chapel name is prefixed with 'chpl_' to make it internal, but the
+  // C type is 'c_string' which is defined in the runtime as an alias for
+  // 'const char*'. The user-facing alias is defined in 'ChapelBase' so that
+  // it can easily be deprecated.
+  // Note that we actually map to the type 'c_string_rehook' to avoid a
+  // collision with the type alias when '--no-munge-user-idents' is thrown.
+  dtStringC                            = createPrimitiveType("chpl_c_string", "c_string_rehook");
```

Comment:
This problem does not make sense to me. It sounds very strange. I would expect that we can have multiple names for the same type that all use the same C type name, without issue.

Reply:
I made a note in the OP as well as updating the comment here, but I don't see a need to resolve it in this PR unless there's something glaring. It makes sense to me why the collision started occurring after we declared `c_string` in the module code and also had given this type the internal cname `c_string`.